File: Java_Codes/7. Exception Handling/Throw.java
class Sample{
    public static void checkNumber(int number) throws IllegalArgumentException {
        if (number < 0) {
            // Manually throw an exception using 'throw'
            throw new IllegalArgumentException("Number must be non-negative.");
        }
        System.out.println("Number is valid: " + number);
    }

    public void Sample1(){
        try {
            // The method call may throw an exception
            checkNumber(-5);
        } 
        catch (IllegalArgumentException e) {
            // The thrown exception is caught here
            System.out.println("Caught Exception: " + e.getMessage());
        }
    }
}


public class Throw {
    public static void main(String[] args) {
        Sample s1 = new Sample();
        s1.Sample1();
    }
}

File: Python_Codes/7. Exception Handling/Throw.py
# Define a class with methods for checking numbers
class Sample:
    @staticmethod
    def check_number(number):
        if number < 0:
            # Manually raise an exception
            raise ValueError("Number must be non-negative.")
        print("Number is valid:", number)

    def sample1(self):
        try:
            # This call may raise an exception
            self.check_number(-5)
        except ValueError as e:
            # Catch the raised exception
            print("Caught Exception:", e)


def main():
    s1 = Sample()
    s1.sample1()


if __name__ == "__main__":
    main()
